src/Disks/IO/AsynchronousBoundedReadBuffer.cpp

```diff
@@ -132,8 +145,9 @@ void AsynchronousBoundedReadBuffer::prefetch(Priority priority)
     last_prefetch_info.submit_time = std::chrono::system_clock::now();
     last_prefetch_info.priority = priority;
 
-    prefetch_buffer.resize(buffer_size);
-    prefetch_future = readAsync(prefetch_buffer.data(), prefetch_buffer.size(), priority);
+    if (!use_page_cache)
+        prefetch_buffer.resize(buffer_size);
```

Comment:
Does this work on condition that we do not use `prefetch_buffer` and `memory` buffer in parallel? I mean if we call `prefetch()` at the end of each `nextImpl` (schedule to get the next data, while we did not get started with reading from just read data pointed to by `memory`) then it will not work with page cache?

Reply:
It works, `working_buffer` will continue to point into `AsynchronousBoundedReadBuffer::page_cache_cell`, while the async reader's `PageCache::getOrSet` will read into another page cache cell (or will hit the same block and produce another shared_ptr to the same cell).

Reply:
Ah, that explains everything, thanks, indeed there is a comment about it in AsynchronouseReader.h, which I did not read, sorry.